tests/src/Interop/MarshalAPI/ReleaseComObject/ReleaseComObjectTest.cs

```diff
@@ -4,83 +4,45 @@
+            Console.WriteLine("ArgumentNullException thrown by ReleaseComObject for null as expected.");
         }
         catch (NullReferenceException)
         {
             // Desktop CLR behavior (test must pass both ProjectN and Desktop CLR behaviors)
-            TestTools.InformationWriteLine("NullReferenceException thrown by ReleaseComObject for null as expected.");
+            Console.WriteLine("NullReferenceException thrown by ReleaseComObject for null as expected.");
         }
         catch (Exception ex)
         {
-            TestTools.ErrorWriteLine("Failed ReleaseComObject test.");
-            TestTools.ErrorWriteLine("Exception occurred: {0}", ex);
+            Assert.Fail("Failed ReleaseComObject test. Unexpected Exception occurred from ReleaseComObject when passed null as parameter: {0}", ex);
         }
 
         try //Test for non-COM object
         {
             Marshal.ReleaseComObject(new object());
 
-            TestTools.ErrorWriteLine("Failed ReleaseComObject test.");
-            TestTools.ErrorWriteLine("No exception from ReleaseComObject when passed non-COM object as parameter.");
+            Assert.Fail("Failed ReleaseComObject test. No exception from ReleaseComObject when passed non-COM object as parameter.");
         }
         catch (ArgumentException)
         {
-            TestTools.InformationWriteLine("ArgumentException thrown by ReleaseComObject for non-COM object as expected.");
-        }
-        catch (Exception ex)
-        {
-            TestTools.ErrorWriteLine("Failed ReleaseComObject test.");
-            TestTools.ErrorWriteLine("Unexpected Exception occurred: {0}", ex);
-        }
-
-        int retValue;
-
-        try
-        {
-            PropertySet propertySet = new PropertySet(); //Create an RCW associated with COM object
-
-            retValue = Marshal.ReleaseComObject(propertySet);
-
-            //Since we typyically only keep one RCW for one object
-            //Marshal.ReleaseComObject should have returned 0
-            if (retValue != 0)
-            {
-                TestTools.ErrorWriteLine("Failed ReleaseComObject test.");
-                TestTools.ErrorWriteLine("Ref count returned from ReleaseComObject is not zero. RefCount: " + retValue);
-            }
-
-            //This should result in an Exception
-            int count = propertySet.Count;
-
-            TestTools.ErrorWriteLine("Failed ReleaseComObject test.");
-            TestTools.ErrorWriteLine("No exception when accessing an object already released by ReleaseComObject.");
-        }
-        catch (InvalidComObjectException)
-        {
-            TestTools.InformationWriteLine("InvalidComObjectException thrown for accessing an object already released by ReleaseComObject as expected.");
+            Console.WriteLine("ArgumentException thrown by ReleaseComObject for non-COM object as expected.");
         }
         catch (Exception ex)
```

Comment:
Catching all exceptions is wrong in this case. If you refer to the `readme.md` in the `Interop/` the `Main` function should catch all exceptions not the tests. We want a single failure point since the tests EXE should not crash due to test failure, it should properly return an error code. Note that `Assert.Fail()` throws an exception.

Reply:
Thanks! This should be fixed.